test/minikube/minikube_domain_resync_test.go

```diff
@@ -0,0 +1,186 @@
+		if ok {
+			assert.Equal(t, "SM", obj.Type, "archive-pvc label should only be present for SMs")
+			// check that PVC exists
+			k8s.RunKubectl(t, options, "get", "pvc", claimName)
+			// add mapping of PVC to archive ID
+			output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+				"nuocmd", "get", "value", "--key", "archiveVolumeClaims/"+claimName)
+			assert.NilError(t, err)
+
+			archiveId, err := strconv.Atoi(strings.TrimSpace(output))
+			assert.NilError(t, err)
+			archiveVolumeClaims[claimName] = archiveId
+		} else {
+			assert.Equal(t, "TE", obj.Type, "archive-pvc label should only be absent for TEs")
+		}
+	}
+	return archiveVolumeClaims
+}
+
+func checkArchives(t *testing.T, namespaceName string, adminPod string, numExpected int, numExpectedRemoved int) (archives []testlib.NuoDBArchive, removedArchives []testlib.NuoDBArchive) {
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+
+	// check archives
+	output, err := k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "archives", "--db-name", "demo")
+	assert.NilError(t, err, output)
+
+	err, archives = testlib.UnmarshalArchives(output)
+	assert.NilError(t, err)
+	assert.Equal(t, numExpected, len(archives), output)
+
+	// check removed archives
+	output, err = k8s.RunKubectlAndGetOutputE(t, options, "exec", adminPod, "--",
+		"nuocmd", "--show-json", "get", "archives", "--db-name", "demo", "--removed")
+	assert.NilError(t, err, output)
+
+	err, removedArchives = testlib.UnmarshalArchives(output)
+	assert.NilError(t, err)
+	assert.Equal(t, numExpectedRemoved, len(removedArchives), output)
+	return
+}
+
+func TestDomainResync(t *testing.T) {
+	testlib.AwaitTillerUp(t)
+	defer testlib.VerifyTeardown(t)
+	defer testlib.Teardown(testlib.TEARDOWN_ADMIN)
+
+	helmChartReleaseName, namespaceName := testlib.StartAdmin(t, &helm.Options{}, 1, "")
+
+	admin0 := fmt.Sprintf("%s-nuodb-cluster0-0", helmChartReleaseName)
+
+	defer testlib.Teardown(testlib.TEARDOWN_DATABASE) // ensure resources allocated in called functions are released when this function exits
+
+	testlib.StartDatabase(t, namespaceName, admin0, &helm.Options{
+		SetValues: map[string]string{
+			"database.sm.resources.requests.cpu":    "0.25",
+			"database.sm.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+			"database.te.resources.requests.cpu":    "0.25",
+			"database.te.resources.requests.memory": testlib.MINIMAL_VIABLE_ENGINE_MEMORY,
+		},
+	})
+
+	originalArchiveVolumeClaims := verifyProcessLabels(t, namespaceName, admin0)
+	assert.Equal(t, 1, len(originalArchiveVolumeClaims))
+	originalArchiveId := -1
+	for _, archiveId := range originalArchiveVolumeClaims {
+		originalArchiveId = archiveId
+	}
+	assert.Assert(t, originalArchiveId != -1)
+
+	// update replica count
+	options := k8s.NewKubectlOptions("", "")
+	options.Namespace = namespaceName
+
+	statefulSets := getStatefulSets(t, namespaceName).Items
```

Comment:
I think that with Helm you have a guarantee that all 3 will exist. Now that I am rewriting the tests for the Helm Operator, I noticed that sometimes the test is faster than the operator and the k8s objects are not yet created. I had to change a bunch of these into Awaits.